application/src/main/java/org/openhds/mobile/utilities/OdkCollectHelper.java

```diff
@@ -149,4 +159,101 @@ public static void setStatusComplete(ContentResolver resolver, Uri uri) {
         resolver.update(uri, cv, null, null);
 
     }
+
+    private static String makePlaceholders(int len) {
+        if (len < 1) {
+            // It will lead to an invalid query anyway ..
+            throw new RuntimeException("No placeholders");
+        } else {
+            StringBuilder sb = new StringBuilder(len * 2 - 1);
+            sb.append("?");
+            for (int i = 1; i < len; i++) {
+                sb.append(",?");
+            }
+            return sb.toString();
+        }
+    }
+
+    public static List<FormInstance> getFormInstancesByPath(ContentResolver resolver, Collection<String> ids) {
+
+        ArrayList<FormInstance> formInstances = new ArrayList<FormInstance>();
+
+        Cursor cursor = resolver.query(CONTENT_URI, new String[]{
+                        InstanceProviderAPI.InstanceColumns.INSTANCE_FILE_PATH,
+                        InstanceProviderAPI.InstanceColumns._ID,
+                        InstanceProviderAPI.InstanceColumns.JR_FORM_ID,
+                        InstanceProviderAPI.InstanceColumns.DISPLAY_NAME,
+                        InstanceProviderAPI.InstanceColumns.STATUS},
+                InstanceProviderAPI.InstanceColumns.INSTANCE_FILE_PATH
+                        + " IN (" + makePlaceholders(ids.size()) + ")",
+                ids.toArray(new String[]{}), null);
+
+        if (null == cursor) {
+            return null;
+        }
+
+        while (cursor.moveToNext()) {
+
+            FormInstance formInstance = new FormInstance();
+            String filePath;
+            String formName;
+            String fileName;
```

Comment:
Just to save you some space and a bit of typing: String filePath, formName, fileName;